library/Makefile

```diff
@@ -1,8 +1,23 @@
 
 # Also see "include/mbedtls/config.h"
 
+ifeq ($(IAR_BUILD),1)
+CFLAGS	?= --arm -Om --silent
+
+# gcc comparable warning flags are not present in IAR.
+# All warnings are enabled by default.
+WARNING_CFLAGS ?=
+
+# IAR archive command does not have ar compatible options 
+ARFLAGS =
```

Comment:
Yes, sorry that was a typo, I mean `?=`. Even if you do this change, the ARFLAGS will be empty if the user does not pass any flags, so I do not see the problem.